### PR TITLE
[expo-updates][ios] Stub expo-updates in Expo Go and remove service pattern

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Expo Updates was using the `EX_REGISTER_MODULE` macro to register a module at runtime that could be overridden in Expo Go, in an effort to swap out nearly the whole core of the updates system at runtime to delegate to the custom updates system in Expo Go for the JS API constants.

This PR inverts this concept by just swapping out the JS module implementation at runtime.

# How

1. Add a separate JS module implementation for expo-updates in Expo Go. This is possible because using updates in Expo Go already doesn't make much sense, but we still want to return somewhat useful constants for the JS API (even if calling the methods is and has always been disallowed).
2. Add a line to the module registry to not register an already-registered module. I'm nearly positive this is incorrect (and probably breaks the core concept of module overriding), but it's tough to comprehend exactly how this system works. Would appreciate reviewer attention to this piece. The issue I'm seeing is that the real module is being registered later than the expo-go-specific one. Maybe we need to create a concept in the registry like `modulesThatShouldOnlyBeRegisteredOnce` that is a list of modules that register checks before registering a duplicate module.
3. Remove the `EX_REGISTER_MODULE` thing and just use the controller directly in the real module implementation. This is the overall goal of this PR as it will allow a refactor of the JS API to go through the controller and use swift rather than an obj-c layer/shim having to be in-between.

# Test Plan

Run an experience with expo-updates installed in Expo Go built locally. Check main branch before this and check this branch and see no differences.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-modules-core/ios/ModuleRegistry.swift [hunk 1]
```diff
@@ -14,6 +14,10 @@ public final class ModuleRegistry: Sequence {
    */
   internal func register(holder: ModuleHolder) {
     log.info("Registering module '\(holder.name)'")
+    // module already registered, don't re-register
+    if registry[holder.name] != nil {
+      return
+    }
```
Comment (wschurman) [inline packages/expo-modules-core/ios/ModuleRegistry.swift:20]: @tsapeta - this is the piece of this PR that needs the most attention. see note in summary.

Comment (douglowder) [reply packages/expo-modules-core/ios/ModuleRegistry.swift:20]: Honestly this seems like it should have been there all along -- or alternatively, throw if the system tries to register the same module twice so we can understand why.

Comment (wschurman) [reply packages/expo-modules-core/ios/ModuleRegistry.swift:20]: The problem is a lot of expo go relies upon overriding modules (I think, unconfirmed if whether it's by this mechanism or another).

**Review by douglowder — APPROVED** [review]

File: packages/expo-modules-core/ios/ModuleRegistry.swift (hunk 1 above)
Comment (tsapeta) [reply packages/expo-modules-core/ios/ModuleRegistry.swift:20]: I think it's fine for now, but could you please elaborate more in the comment above (specifically mention that it's necessary for expo-updates and Expo Go) and maybe do `log.warn` that says that the module is already registered and was not overridden.

In the future, we can think what we can do to make it better. Maybe the `register(module:)` could take another param like `canBeOverridden` that we set in the ModuleHolder and then skip re-registering when it's set to `false` (`true` by default)

**Review by tsapeta — APPROVED** [review]

File: packages/expo-modules-core/ios/ModuleRegistry.swift (hunk 1 above)
Comment (wschurman) [reply packages/expo-modules-core/ios/ModuleRegistry.swift:20]: I think I'll make that `canBeOverridden` change now to reduce the possible scope of this PR.

**Pull request merged**